### PR TITLE
reduced chat delay

### PR DESCRIPTION
**Pull request by KentoNishi**
I figured out a trick to make the delay negligible.

When a chunk is received and is completely sorted, we assume that the first message in the chunk is meant to be displayed immediately. All messages that follow are timed relative to the first message. I think this is what ytc does internally, since the timing is very very close to default ytc. So much so that screenshots in chilledcow chat show an identical list of messages

left hc, right ytc.

image

### CODE REVIEW

**Comment by KentoNishi** [comment]
converting pr to draft cuz i realized i might have to change things for replay chat

**Comment by KentoNishi** [comment]
image

this is pog

File: src/ts/chat-parser.ts [hunk 1]
```diff
@@ -209,6 +209,13 @@ const sortAction = (action: Ytc.ParsedAction, messageArray: Ytc.ParsedMessage[],
   }
 };
 
+const cheatTimestamps = (arr: Ytc.ParsedMessage[]): void => {
+  const earliest = arr[0].showtime;
+  arr.forEach(item => {
+    item.showtime += Date.now() - earliest;
+  });
+};
```
Comment (ChrRubin) [inline src/ts/chat-parser.ts:217]: Can calculate the diff before forEach so `Date.now()` won't get called for each message.

**Review by ChrRubin — CHANGES_REQUESTED** [review]
Sometimes the easiest solution is the best solution lmao

**Review by ChrRubin — APPROVED** [review]

**Pull request merged**